tests/MongoDB.Driver.Tests/MongoDatabaseImplTests.cs

```diff
@@ -482,6 +484,106 @@ public void Settings_should_be_set()
             op.WriteConcern.Should().BeSameAs(writeConcern);
         }
 
+        [Theory]
+        [ParameterAttributeData]
+        public void ListCollectionNames_should_execute_a_ListCollectionsOperation(
+            [Values(false, true)] bool usingSession,
+            [Values(false, true)] bool async)
+        {
+            var session = CreateSession(usingSession);
+            var filterDocument = BsonDocument.Parse("{ name : \"awesome\" }");
+            var filterDefinition = (FilterDefinition<BsonDocument>)filterDocument;
+            var options = new ListCollectionNamesOptions
+            {
+                Filter = filterDefinition
+            };
+            var cancellationToken = new CancellationTokenSource().Token;
+
+            var mockCursor = new Mock<IAsyncCursor<BsonDocument>>();
+            _operationExecutor.EnqueueResult<IAsyncCursor<BsonDocument>>(mockCursor.Object);
+
+            if (usingSession)
+            {
+                if (async)
+                {
+                    _subject.ListCollectionNamesAsync(session, options, cancellationToken).GetAwaiter().GetResult();
+                }
+                else
+                {
+                    _subject.ListCollectionNames(session, options, cancellationToken);
+                }
+            }
+            else
+            {
+                if (async)
+                {
+                    _subject.ListCollectionNamesAsync(options, cancellationToken).GetAwaiter().GetResult();
+                }
+                else
+                {
+                    _subject.ListCollectionNames(options, cancellationToken);
+                }
+            }
+
+            var call = _operationExecutor.GetReadCall<IAsyncCursor<BsonDocument>>();
+            VerifySessionAndCancellationToken(call, session, cancellationToken);
+
+            var op = call.Operation.Should().BeOfType<ListCollectionsOperation>().Subject;
+            op.DatabaseNamespace.Should().Be(_subject.DatabaseNamespace);
+            op.Filter.Should().Be(filterDocument);
+            op.NameOnly.Should().BeTrue();
+        }
+
+        [SkippableTheory]
+        [ParameterAttributeData]
+        public void ListCollectionNames_should_return_expected_result(
```

Comment:
We don't usually write integration tests for High but since ListCollectionNames does further processing on the result of the command in runs we need to.